Formula/benerator.rb

```diff
@@ -0,0 +1,116 @@
+class Benerator < Formula
+  desc "Tool for realistic test data generation"
+  homepage "https://rapiddweller.github.io/homebrew-benerator/"
+  url "https://github.com/rapiddweller/rapiddweller-benerator-ce/releases/download/3.1.0/rapiddweller-benerator-ce-3.1.0-jdk-11-dist.tar.gz"
+  sha256 "194feb051ae18cfcd407b8e1668ce9c60561394bc454f9fc9747c274166843bc"
+
+  license "Apache-2.0"
+  depends_on "openjdk"
+
+  def install
+    # Remove unnecessary files
+    rm_f Dir["bin/*.bat", "bin/pom.xml"]
+
+    # Installs only the "bin" and "lib" directories from the tarball
+    libexec.install Dir["bin", "lib"]
+    # Generate a script that sets the necessary environment variables
+    env_script = libexec / "benerator-env"
+    env_script.write_env_script(
+      libexec / "bin/benerator",
+      JAVA_HOME:      Formula["openjdk"].opt_prefix,
+      BENERATOR_HOME: libexec,
+    )
+
+    # Install the wrapper script
+    bin.install libexec / "benerator-env" => "benerator"
+    rm_f "#{bin}/benerator_common"
+    rm_f "#{bin}/log4j2.xml"
+  end
+
+  def caveats
+    <<~EOS
+      To use the benerator commands, please set the following environment variables:
+
+      BENERATOR_HOME="#{libexec}"
+      JAVA_HOME="$(/usr/libexec/java_home)"
+
+    EOS
+  end
+
+  test do
+    # Test if version is correct
+    ohai "Test if the correct version of benerator is installed"
+    assert_match "Benerator Community Edition 3.1.0-jdk-11",
```

Comment:
```suggestion
    assert_match "Benerator Community Edition #{version}-jdk-#{Formula["openjdk"].version.major}",
```

Reply:
update in this https://github.com/Homebrew/homebrew-core/pull/124749#discussion_r1129808260 with new test also